### ISSUE TITLE
False positive in javascript.lang.security.audit.non-constant-sql-query.non-constant-sql-query

### ISSUE THREAD

**Issue by sk-** [returntocorp/sgrep-rules#1281]
**Describe the bug**
The pattern for `javascript.lang.security.audit.non-constant-sql-query.non-constant-sql-query` is too broad, as it will complain for any call to a `query` method which does not receive a string. For example when mocking requests using `nock`.

**To Reproduce**
```js
nock('https://api.url.com')
    .get('/endpoint')
    .query({ limit: '100' })
```

**Expected behavior**
The rule should not complain. It should either detect that no sql is being executed in this module or detect that the argument is not a templated string.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [X] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

**Comment by inkz** [comment]
Thanks again @sk-!
commit: https://github.com/returntocorp/semgrep-rules/pull/1321
